launch/org.eclipse.cdt.docker.launcher/src/org/eclipse/cdt/internal/docker/launcher/ContainerPropertyTab.java

```diff
@@ -170,83 +173,140 @@ public void createControls(Composite parent) {
 
 		usercomp.setLayoutData(gd);
 
-		enableButton = new Button(usercomp, SWT.CHECK);
-		enableButton.setText(Messages.ContainerPropertyTab_Enable_Msg);
+		//Enable button
+		{
```

Comment:
Not sure why you are breaking up the code with blocks.  Comments are ample.

Reply:
FWIW I am fond of these extra blocks, I would go further an locally declare stuff like gd in each block as that makes it easier to detect when a gd is being reused mistakenly (especially when stuff is copied/pasted in the future). However to me none of these things are blocking changes and as @cmorty is actively maintaining this, I am pleased to see it in a style he enjoys reading and that @jjohnstn has merged these changes.

Reply:
@jonahgraham : Yes. I added the blocks because I lost orientation. As for the gd: I thought about that, but then was to lazy. ;) I first had to learn, that I'm not able to reuse gd, which IMO is a bummer. :-)

Reply:
Thanks @cmorty for the comment. I too remember having to learn that you can't reuse gd.  The other way to do this is to not make a `gd` variable at all and use the factory, something like this:

```java
GridDataFactory.fillDefaults().grab(false, false).applyTo(label);
```

Its not really worth changing old code to that style, but it is useful to know for new code.

See [`GridDataFactory` javadoc](https://help.eclipse.org/latest/topic/org.eclipse.platform.doc.isv/reference/api/org/eclipse/jface/layout/GridDataFactory.html) for more examples of how to use it.